OpenAdaptAI/OpenAdapt
# Add ci scripts to run the app after build

**What kind of change does this PR introduce?**
This PR introduces simple scripts that runs the built app in the CI and checks if the app is able to run without issues



**Summary**
This PR adds CI scripts that download the built apps for each platform, run them, and after an interval, check if the processes are still running. The app code is designed such that any exceptions, the app will exit gracefully, and so the script will not be able to find the process running, and accordingly raise the error.

- https://github.com/KIRA009/OpenAdapt/actions/runs/9367703753 Run in MacOS with a valid build
- https://github.com/KIRA009/OpenAdapt/actions/runs/9368019318 Run in MacOS, with an invalid build (https://github.com/KIRA009/OpenAdapt/commit/b5235e2e914713177c92f372db7ac75886b82402#diff-c79b1569d910854377b3fb9d79e20d60bfe7593aab5a7444f7c5ef1aa616a121)
- https://github.com/KIRA009/OpenAdapt/actions/runs/9369173544 Run in Windows, with a valid build
- https://github.com/KIRA009/OpenAdapt/actions/runs/9368935768 Run in Windows, with an invalid build (https://github.com/KIRA009/OpenAdapt/commit/0b2adf0f494bffce94742c3a9f9d963d0c0357a5#diff-c79b1569d910854377b3fb9d79e20d60bfe7593aab5a7444f7c5ef1aa616a121)



**Checklist**

**How can your code be run and tested?**


You can create an invalid build (by raising Exceptions, or commenting out packages from the `build.py`) and push to your fork (make sure to change the `on` condition on `release-and-publish.yml` to any branch (`**`))
